kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml

```diff
@@ -1680,36 +1685,24 @@ components:
         messagesBehind:
           type: integer
           format: int64
+        consumerId:
+          type: string
+        host:
+          type: string
       required:
-        - consumerId
+        - topic
+        - partition
 
-    TopicConsumerGroups:
-      type: object
-      properties:
-        consumers:
-          type: array
-          items:
-            $ref: '#/components/schemas/ConsumerTopicPartitionDetail'
 
     ConsumerGroupDetails:
-      type: object
-      properties:
-        consumerGroupId:
-          type: string
-        simple:
-          type: boolean
-        partitionAssignor:
-          type: string
-        state:
-          type: string
-        coordintor:
-          type: string
-        consumers:
-          type: array
-          items:
-            $ref: '#/components/schemas/ConsumerTopicPartitionDetail'
-      required:
-        - consumerGroupId
+      allOf:
+        - $ref: '#/components/schemas/ConsumerGroup'
+        - type: object
+          properties:
+            partitions:
+              type: array
+              items:
+                $ref: '#/components/schemas/ConsumerGroupTopicPartition'
```

Comment:
We've got an issue here. `ConsumerGroup` type has the `state` field which is an enum. And for some weird reason, OpenAPI generates two different Typescript enums for `ConsumerGroup` and `ConsumerGroupDetails` - `ConsumerGroupStateEnum` and `ConsumerGroupDetailsStateEnum` respectively. So, although de facto these enums are identical, the compiler treats them like two different types, which breaks a lot of stuff on the frontend. Can we somehow force these fields to be of the same type?

Reply:
Previously, they both were `string`s, so it worked. Not anymore

Reply:
I guess, this can be solved by abstracting this enum to a separate type and using it by reference, but I'm not sure.

Reply:
Yes, let's move it to separate class